### DISCUSSION TITLE
contour filter algorithm

### DISCUSSION THREAD

**Discussion by danpak94** [pyvista/pyvista#4461] [Q&A] [answered]
Hello,

I have been using the ```contour``` filter quite frequently to generate surface meshes from SDF.

From what I understand, there are three options for the contour method - "contour", "marching_cubes", and "flying_edges". Marching cubes and flying edges have descriptive names so I could look up the associated algorithms, but what algorithm is behind "contour"?

I tried to find the answer in the vtk documentation (```vtkContourFilter```) and other pages, but could not find the info. The only thing remotely close was "edge-tracking" mentioned in this [page](https://examples.vtk.org/site/VTKBook/06Chapter6/#contouring), but I wasn't sure if this was referring to ```vtkContourFilter```.

I would appreciate any help or insight from the community.

Thank you

**Comment by tkoyama010** [comment, marked as answer]
Thank you for initiating this thread. As you may be aware, PyVista primarily wrapping vtk's vtkContourFilter. Therefore, for a more precise response regarding the internal algorithm, it would be beneficial to post your query on the [VTK Forum](https://discourse.vtk.org/).

**Reply by danpak94** [reply]
Hello Tetsuo,

Thank you for the suggestion. I will ask the same question in the VTK forum instead.
